plugin/path_checkout_sets.go

```diff
@@ -316,23 +334,54 @@ func storeSet(ctx context.Context, storage logical.Storage, setName string, set
 	return storage.Put(ctx, entry)
 }
 
+func (b *backend) checkInNewServiceAccount(ctx context.Context, storage logical.Storage, serviceAccountName string) (isUserErr bool, err error) {
+	lock := locksutil.LockForKey(b.checkOutLocks, serviceAccountName)
+	lock.Lock()
+	defer lock.Unlock()
+
+	_, err = b.checkOutHandler.Status(ctx, storage, serviceAccountName)
+	if err == nil {
+		// We actually want to receive ErrNotFound here because that would indicate
+		// we're not already managing the potential to check this service account out
+		// through another set.
+		return true, fmt.Errorf("%s is already managed by another set, please remove it and try again", serviceAccountName)
+	}
+	if err != ErrNotFound {
+		// This is probably a persistent issue with reaching storage.
+		return false, err
+	}
+
+	// All is well.
+	// Check in the service account so it'll be listed as managed by this
+	// plugin and available.
+	if err := b.checkOutHandler.CheckIn(ctx, storage, serviceAccountName); err != nil {
+		return false, err
+	}
+	// Success.
+	return false, nil
+}
+
 // deleteSetServiceAccount errors if an account can't presently be deleted, or deletes it.
-func (b *backend) deleteSetServiceAccount(ctx context.Context, storage logical.Storage, serviceAccountName string) error {
+func (b *backend) deleteSetServiceAccount(ctx context.Context, storage logical.Storage, serviceAccountName string) (isUserErr bool, err error) {
```

Comment:
They're both in use if you take a second look. I would link you to them directly but the line they're on is likely to shift with the edits I'll push with this PR.

Reply:
I'm not seeing where `isUserErr` is used either, nor any naked `return`, so I guess I have the same question about the named result parameters.

Reply:
FWIW this is no longer relevant. With making the endpoints no longer do partial updates, `isUserErr` was no longer needed because I just return immediately.